client/src/modules/journal/journal.js

```diff
@@ -510,19 +531,33 @@ function JournalController(
   //       updated to keep track of the current grid grouped state
   vm.gridGroupedState = vm.grouping.getCurrentGroupingColumn;
 
+  // initial loading in startup() function
+  function initLoading() {
+    load(Journal.filters.formatHTTP(true));
+    vm.latestViewFilters = Journal.filters.formatView();
+  }
   // runs on startup
   function startup() {
     var hasStateFilters = $state.params.filters.length > 0;
+    var noCachaleKeys = [];
     if (hasStateFilters) {
       Journal.filters.replaceFiltersFromState($state.params.filters);
+
+      initLoading();
+      for (var i = 0; i < $state.params.filters.length; i++) {
```

Comment:
Hmm... this works, but we could use this functionality across the entire application if we made it a feature of the the `FilterService`.

I propose that you instead modify the [FilterService.prototype.formatCache()](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/FilterService.js#L130) function to only cache filters that are `cacheable`.  I imagine the code will look something like this:
```js
FilterList.prototype.formatCache = function formatCache() {
  var filters = {};

   // loop through each filter and only include it if it is cacheable
  angular.forEach(this._filterIndex, function (filter, key) {
     if (filter.cacheable === 0) { return; }
     filters[key] = filter;
  });

  return filters;
}
```

Reply:
That way, you actually don't have to change any code in the `Journal` and other modules will immediately benefit by simply adding `cacheable: 0` to their filter parameters.

Reply:
cool, but 
```js 
this._filterIndex  
``` 
does not save cachable index,
Another problem,  data should be loaded with all parameters for the first time and we can remove uncachable filters after
I created this function instead
```js
 // uncachable filters should not remain for a long time, so they can be remove if needed
  FilterList.prototype.removeUnCachableFilters = function removeUnCachableFilters(filters) {
    filters.forEach(filter => {
      if (filter.cachable === 0) {
        delete this._filterIndex[filter.key];
      }
    });
  };